syntax = "proto3";

package ace;

option go_package = "pb/proto";

service TutorService {
  rpc ProcessSyllabus (SyllabusRequest) returns (SyllabusResponse);
  rpc Chat (ChatRequest) returns (ChatResponse);
}

message SyllabusRequest {
  string file_name = 1;
  bytes file_data = 2; // NEW: The actual PDF bytes
}

message SyllabusResponse {
  bool success = 1;
  string message = 2;
  int32 nodes_created = 3;
  string graph_json = 4; // NEW: The node data for the UI
}

message ChatRequest {
  string message = 1;      // The user's question
  string session_id = 2;   // To track conversation history later
}

message ChatResponse {
  string response = 1;     // The AI's answer
}